codec-http/src/main/java/io/netty/handler/codec/http/websocketx/WebSocketClientProtocolHandshakeHandler.java

```diff
@@ -70,6 +70,12 @@ public void operationComplete(ChannelFuture future) throws Exception {
         applyHandshakeTimeout();
     }
 
+    @Override
+    public void channelInactive(ChannelHandlerContext ctx) throws Exception {
+        handshakePromise.tryFailure(new WebSocketHandshakeException("channel closed with handshake in progress"));
```

Comment:
Lets first check if it’s done yet and only if this is not true try to fail it. This will reduce the overhead in most cases as we will not create the extra exception

Reply:
Done.